### PR TITLE
Proposal: Fields mapping

### PR DESCRIPTION
**Pull request by dkulyk**

### CODE REVIEW

**Comment by dkulyk** [comment]
Maybe we should decode JSON to an array? 
And prepare objects before returning the result?

**Comment by IsraelOrtuno** [comment]
This looks good, could you add some testing?

File: src/Resources/Notes.php [hunk 1]
```diff
@@ -3,9 +3,11 @@
 namespace Devio\Pipedrive\Resources;
 
 use Devio\Pipedrive\Resources\Basics\Resource;
+use Devio\Pipedrive\Resources\Traits\FieldsMapping;
 
 class Notes extends Resource
 {
+    use FieldsMapping;
```
Comment (IsraelOrtuno) [inline src/Resources/Notes.php]: A note does not have dynamic fields on Pipedrive

File: src/Resources/Activities.php [hunk 2]
```diff
@@ -4,7 +4,9 @@
 
 use Devio\Pipedrive\Resources\Basics\Resource;
 use Devio\Pipedrive\Resources\Traits\DeletesInBulk;
+use Devio\Pipedrive\Resources\Traits\FieldsMapping;
 
 class Activities extends Resource
 {
+    use FieldsMapping;
```
Comment (IsraelOrtuno) [inline src/Resources/Activities.php]: Same here, an activity does not have EAV, do it?

File: src/Resources/Notes.php (hunk 1 above)
Comment (dkulyk) [inline src/Resources/Notes.php]: I confused the name NoteFields. ActivityFields too.

**Comment by IsraelOrtuno** [comment]
Looks pretty good, let's add some testing and go :D thanks dude!

**Comment by IsraelOrtuno** [comment]
@dkulyk can you add testing and I merge? ;D

File: src/Resources/Basics/Resource.php [hunk 3]
```diff
@@ -130,25 +156,26 @@ public function isEnabled($method)
...
+        // First we will make sure the method only belongs to this abstract class
+        // as this does not have to interfere with methods described in child
         // classes. We can now check if it is found in the enabled property.
-        if (! in_array($method, get_class_methods(get_class()))) {
+        if (!in_array($method, get_class_methods(get_class()), true)) {
             return true;
         }
 
-        return in_array($method, $this->enabled) || $this->enabled == ['*'];
+        return count($this->enabled) === 0 || in_array($method, $this->enabled, true);
```
Comment (IsraelOrtuno) [inline src/Resources/Basics/Resource.php]: Not really sure of this, what about `*` support? Dropped?

**Pull request closed without merging**